django/src/rdwatch/management/commands/ingest.py

```diff
@@ -0,0 +1,272 @@
+                            "data": data,
+                            "size": size,
+                            "nodata_value": bnd.nodata_value,
+                        }
+                    ],
+                }
+            )
+
+
+def _ingest_geojson(
+    site_geojson_file: Path,
+    prediction_configuration: PredictionConfiguration,
+    tracking_configuration: TrackingConfiguration,
+) -> list[Site]:
+    """
+    Ingest the geojson file at the given path into the Site model,
+    returning a list of Site instances.
+    """
+    site_geojson = json.loads(site_geojson_file.read_text())
+
+    sites: list[Site] = []
+
+    # Ingest each site in the geojson
+    for i, site in enumerate(site_geojson["features"]):
+        properties: dict = site["properties"]
+
+        # The first item in the array, the ground truth, doesn't have an
+        # explicit label.
+        if i == 0:
+            assert "current_phase" not in properties
+            label = Site.GROUND_TRUTH
+        else:
+            label = label_mapping[properties["current_phase"]]
+
+        score = properties["score"]
+
+        geometry = GEOSGeometry(json.dumps(site["geometry"]))
+        assert isinstance(geometry, MultiPolygon | Polygon)
+
+        # Generate a MultiPolygon if the geometry is just a Polygon (this
+        # seems to be the case for all ground truth features).
+        if isinstance(geometry, Polygon):
+            geometry = MultiPolygon(geometry)
+
+        sensor_name = properties.get("sensor_name")
+
+        if not sensor_name:
+            assert label == Site.GROUND_TRUTH
+            saliency = None
+        else:
+            assert sensor_name
+
+            sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
+
+            satellite_timestamp = properties.get(
+                "observation_date", properties.get("start_date")
+            )
+            assert satellite_timestamp is not None
+
+            satellite_image = SatelliteImage.objects.get_or_create(
+                sensor=sensor_name, timestamp=satellite_timestamp
+            )[0]
+
+            saliency = Saliency.objects.create(
+                configuration=prediction_configuration,
+                source=satellite_image,
+            )
+
+            # Parse the crop string "identifier" from the parent
+            # directory of the source TIF given in the geojson.
+            crop_string = Path(properties["source"]).parent.name
+
+            # From this, we can derive the filepath of the saliency TIF:
+            tif_filepath = (
+                site_geojson_file.parents[3]
+                / "_assets"
+                / "pred_saliency"
+                / f"{crop_string}_salient.tif"
+            )
+
+            # TODO: not sure why some crop strings don't have
+            # a corresponding saliency TIF; there's quite a few in
+            # the sample data that don't. If one is encountered,
+            # just skip it and move on to the next raster.
```

Comment:
Awesome, thanks for spotting this. Either:
1. Our view of the data model is incorrect (each `Site` must be derived from a `Saliency`)
2. We're missing data

I'm banking on number 2, but I'll consult the WATCH team to make sure. For now, logging this is and continuing on the error like you've done is perfect.